frontend.md

```diff
@@ -114,25 +114,185 @@ Object.keys(person).forEach(lang.hitch(this, function(prop){
+  descendent elements, and modifications.
+* It keeps the specificity of selectors low.
+* It helps to decouple presentation semantics from document semantics.
+
+Your components should each be defined in their own SASS file.
+
+#### Namespace
+
+Components can (and probably should) be prefixed with a namespace. For example, you may wish to avoid the potential for collisions between libraries and your custom components by prefixing all your components with a namespace.
 
 ```sass
-.country
-  @include border-bottom-radius(50px)
+.fa-Button
+  /* … */
+.fa-Tabs
+  /* … */
+```
+
+This makes it clear, when reading the HTML, which components are part of your
+library.
+
+<a name="ComponentName"></a>
+#### ComponentName
+
+The component's name must be written in pascal case. Nothing else in the
+HTML/CSS uses pascal case.
+
+```sass
+.fa-MyComponent
+  /* … */
+```
+
+```html
+<article class="fa-MyComponent">
+  …
+</article>
+```
+
+<a name="ComponentName--modifierName"></a>
+#### ComponentName--modifierName
+
+A component modifier is a class that modifies the presentation of the base
+component in some form (e.g., for a certain configuration of the component).
+Modifier names must be written in camel case and be separated from the
+component name by two hyphens. The class should be included in the HTML _in
+addition_ to the base component class.
+
+```sass
+/* Core button */
+.fa-Button
+  /* … */
+/* Cancel and Default buttons often differentiate themselves visually from normal buttons. This would be a perfect usage of --modifierName */
+/* Default button style */
+.fa-Button--default
+  /* … */
+/* Cancel button style */
+.fa-Button--cancel
+  /* … */
+
+/* Core sidebar button */
+.fa-SidebarButton
+  /* … */
+/* Sidebar primary action button style */
+.fa-SidebarButton--primary
+  /* … */
+/* Sidebar secondary action button style */
+.fa-SidebarButton--secondary
+  /* … */
+```
+
+```html
+<button class="fa-Button fa-Button--default" type="button">…</button>
+
+<div class="fa-Sidebar">
+  <button class="fa-Button fa-Button--sidebarPrimary" type="button">
+    Save
+  </button>
+  <button class="fa-Button fa-Button--sidebarSecondary" type="button">
+    Print
+  </button>
+</div>
 ```
 
+<a name="ComponentName-descendentName"></a>
+### ComponentName-descendentName
+
+A component descendent is a class that is attached to a descendent node of a
+component. It's responsible for applying presentation directly to the
+descendent on behalf of a particular component. Descendent names must be
+written in camel case.
+
+```html
+<article class="fa-Tweet">
+  <header class="fa-Tweet-header">
+    <img class="fa-Tweet-avatar" src="{{src}}" alt="{{alt}}">
+    …
+  </header>
+  <div class="fa-Tweet-bodyText">
+    …
+  </div>
+</article>
+```
+
+<a name="is-stateOfComponent"></a>
+### ComponentName.is-stateOfComponent
+
+Use `is-stateName` to reflect changes to a component's state. The state name
```

Comment:
I'm getting caught on using two names for the same thing. Is it possible to clarify the intentions of both in text instead of in name and use just one name?